### PR TITLE
Geometry: Add Triangle Area

### LINKED ISSUE #4286: Geometry: Add Triangle Area
I saw [this issue](https://github.com/duckduckgo/zeroclickinfo-goodies/issues/4216) about adding Geometry support for trapezoids and hexagons. I figured I would add Triangle area as well - even though it's a much easier formula and might get less search volume, I was still surprised it wasn't in the Geometry IA.

This is my first IA contribution, which is very exciting! I'm working on the PR now and will try to submit it soon.

------
IA Page:  http://duck.co/ia/view/geometry 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @pixunil

### PR DESCRIPTION
**Pull request by levthedev**
## Description
This PR adds support for triangle area in the Geometry Instant Answer. I also added "of an" as a trigger word, because I noticed that only "of a" was included. This meant that searching "area of an equilateral triangle" didn't trigger anything, but "area of a equilateral triangle" did. Presumably this is because "equilateral triangle" is the only shape right now that starts with a vowel, but I figure it couldn't hurt (others that this will be relevant for when they have a Geometry IA - oval, ellipse, octagon).

I added 2 tests - one for equilateral triangle and one for triangle. I also made sure that the `duckpan query` and `duckpan server`, and `duckpan test` looked good.

Here is the relevant IA - https://duck.co/ia/view/geometry.

## Related Issues and Discussions
Fixes #4286 

## People to notify
Maintainer of Geometry: @pixunil

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/Geometry

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by adityatandon007** [comment]
@levthedev great start :+1: I suppose you are still working on this. You can add the perimeter too for the traingle.

**Comment by levthedev** [comment]
@adityabhushan - I actually thought that I might leave perimeter out for now. When you have any combination of 3 sides/angles you can solve for all the rest, meaning, there are 6 different formulas, so we'd have to figure out which one to display to the user (or all 6? or a link to a site with them, like WolphramAlpha?).

Of course, we could just say that the perimeter of a triangle is a + b + c, but that assumes you know all three sides and so I imagine is not as useful for most people.

I think ideally it would be more interactive - the user would put in whatever sides or angles they had and it would solve it for them. That's probably out of the scope of this PR, but maybe I should file an issue for an interactive geometry calculator.

Thoughts?

**Comment by adityatandon007** [comment]
@levthedev you spelled my handle name wrongly, nothing  to worry for that. I agree with suggestion regarding making a geometry calculator but I think we should still add that perimeter because we have a trigger word  for perimeter so if u don't add the perimeter somebody typing perimeter of triangle will trigger the answer but they will only see result for area which would annoy the user. I am also familiar with the fact we have some shapes that doesn't show the perimeter but trigger on perimeter. We could work it out together for those shapes too, in fact I  am working on adding more geometries.

I was also thinking for that interactive geometry calc but for this IA it's out of reach. Lets make this one for generic geometry shapes and make a separate IA for geometry calc

**Comment by levthedev** [comment]
@adityatandon007 so sorry about the name mixup!

I just added perimeter for triangles, let me know if there's anything else needed for this PR.

And yeah, I'd love to help out on an interactive geometry calculator as a separate IA.

File: t/Geometry.t [hunk 1]
```diff
@@ -78,7 +78,8 @@ ddg_goodie_test(
     'area of a triangle' => build_test(
         'triangle',
         {
-            area =>      { color => "#F1A031", html => "bh/2", nameCaps => "Area", symbol => "A" }
+            area =>      { color => "#F1A031", html => "bh/2", nameCaps => "Area", symbol => "A" },
+            perimeter => { color => "#5B9E4D", html => "a+b+c",   nameCaps => "Perimeter",  symbol => "u" }
         },
```
Comment (adityatandon007) [inline t/Geometry.t]: one minor nit, just remove the extra space before `nameCaps`

File: t/Geometry.t [hunk 2]
```diff
@@ -78,7 +78,8 @@ ddg_goodie_test(
     'area of a triangle' => build_test(
         'triangle',
         {
-            area =>      { color => "#F1A031", html => "bh/2", nameCaps => "Area", symbol => "A" }
+            area =>      { color => "#F1A031", html => "bh/2", nameCaps => "Area", symbol => "A" },
+            perimeter => { color => "#5B9E4D", html => "a+b+c",   nameCaps => "Perimeter",  symbol => "u" }
```
Comment (adityatandon007) [inline t/Geometry.t]: one minor nit, just remove the extra space before `nameCaps`

**Comment by levthedev** [comment]
@adityatandon007 OK, I've added the height dotted line to the SVG and removed the extra whitespace

**Comment by levthedev** [comment]
Hey, just checking in about this PR. It seems that it's still not on the Beta server.

I know that something was up with the Beta server when I submitted this PR, so I'm wondering if there is a way to manually trigger it to deploy again? Maybe the easiest thing to do is for me to add a junk commit (assuming it attempts deploys on commits)?

Also, GitHub says that merging is blocked because it lacks an approved review, but it looks like I have a review from @adityatandon007. Does that not count as an approved review? I thought it would because @adityatandon007 is the assignee and one of the two reviewers. 

Pinging @nalinbhardwaj because it seems you are one of the other reviewers (do I need both reviewers to approve before it deploys?)

**Comment by adityatandon007** [comment]
@levthedev sorry buddy I forgot about this PR. I don't have write access. I will ask someone to get this deployed to beta and thanks for reminding me of this PR.

**Comment by adityatandon007** [comment]
@pjhampton can you merge this PR.

**Comment by mintsoft** [comment]
Thanks for the work @levthedev apologies this took so long

**Comment by bsstoner** [comment]
This was released to production 🚀

**Comment by mintsoft** [comment]
Awesome to see this happening so quickly @bsstoner !

**Pull request merged**